PP-426/review-documentation (#1)
franciscotobar/devportal: rif/relay/versions.md

In V0.2 implementation, the RelayRequest and the DeployRequest include a relayWorker attribute to identify which account paid for the gas. The RIF Relay SmartWallet pays directly to this account the number of tokens negotiated for the Relay (or Deploy) service. In V1 The relayWorker attribute was removed from the RelayRequest and DeployRequest. A new attribute called feesReceiver was implemented and configured in the RelayServer

This change did not alter the current relay flow, keeping its cost as it is today, and also introduced the flexibility to implement any revenue-sharing strategy needed.

* The feesReceiver could be the worker or MultSig contract. 
* The SmartWallet will pay to the feesReceiver. The feesReceiver will hold the funds of each user payment.
* Upon payment from the SmartWallet, the feesReceiver will not perform any distribution logic to avoid increasing the cost of the relay service for the user.
* With this approach, no changes in the relay flow are required, and thus the introduction of a revenue-sharing mechanism will not impact the price of the relay service.
* The relevant participants will form part of the MultiSig contract.

>     *  The MultiSig contract specify how much of the funds collected go to each participant (e.g., the relayServer operator, the wallet provider, the liquidity provider, and IOVlabs could be the participants).

Reviewer: ```suggestion
    *  The MultiSig contract specify how much of the funds collected go to each participant (e.g., the relayServer operator, the wallet provider, and the liquidity provider could be the participants).
```

Author: updated

    * The maximum gas estimation in the RIF Relay Server is more precise now.
    * A new utility function is available to estimate the maximum gas a relay transaction would consume, based in a linear fit estimation. This can be used in applications that don't want to sign a payload each time they need an approximation of the cost of relaying the transaction.
* Paymaster verifications are done off-chain to optimize gas costs, thus the paymasters are now called Verifiers and they are not part of the on-chain relay flow nor do they handle payments at all.
* Gas cost optimization.
* Security issues.

## Version 1

### Overview

> Including a revenue-sharing mechanism to the RIF Relay service should not introduce a price penalty to the RIF Relay users. With this restriction in mind, we proposed to modify the address used to pay (in tokens) for a successful transaction relay (or deploy). 

Reviewer 2: ```suggestion
Including a revenue-sharing mechanism to the RIF Relay service doesn't introduce a price penalty to the RIF Relay users. We modified the address used to receive tha payment (in tokens) for a successful transaction relay (or deploy).
```

Author: updated

> In its current implementation, the RelayRequest and the DeployRequest include a relayWorker attribute to identify which account paid for the gas. The RIF Relay SmartWallet pays directly to this account the number of tokens negotiated for the Relay (or Deploy) service. The relayWorker attribute can be replaced with the address of a Smart Contract containing the revenue sharing logic. 

Reviewer 2: ```suggestion
In its current implementation, we removed the relayWorker attribute from the RelayRequest and DeployRequest to include a `feesReceiver` attribute in the RelayData. The RIF Relay SmartWallet pays directly to this account the number of tokens negotiated for the Relay (or Deploy) service. The relayWorker attribute can be replaced with the address of a Smart Contract containing the revenue sharing logic.
```

Author: We need to discuss this, since I think that I think in a different approach

Reviewer 2: We agreed on specifying the implementation we are referring to just to clarify that when we say in "in its current implementation" we refer to version 0.2

Author: updated

This change will not alter the current relay flow, keeping its cost as it is today, and it will also introduce the flexibility to implement any revenue-sharing strategy needed.

> * At the moment, there’s no enforced logic on what address the relayWorker attribute from a RelayRequest or DeployRequest can be. We propose for it to be a MultiSig contract address, with all the fee distribution logic. 

Reviewer 2: This is not clear to me

Author: The approach is explained here

> * The relayWorker attribue will be removed from the RelayRequest and DeployRequest. A new attribute called feesReceiver will be implemented and configured in the RelayServer:

Reviewer 2: ```suggestion
* The relayWorker attribute has been removed from the RelayRequest and DeployRequest. A new attribute called feesReceiver will be implemented and configured in the RelayServer:
```

Author: updated

### Overview

> Including a revenue-sharing mechanism to the RIF Relay service doesn't introduce a price penalty to the RIF Relay users. We modified the address used to receive tha payment (in tokens) for a successful transaction relay (or deploy).

Reviewer 2: ```suggestion
Including a revenue-sharing mechanism to the RIF Relay service doesn't introduce a price penalty to the RIF Relay users. We modified the address used to receive the payment (in tokens) for a successful transaction relay (or deploy).
```

my fault, sorry

Author: updated
